test/system/helpers.network.bash

```diff
@@ -0,0 +1,282 @@
+# -*- bash -*-
+
+
+### Feature Checks #############################################################
+
+# has_ipv4() - Check if one default route is available for IPv4
+function has_ipv4() {
+    [ -n "$(ip -j -4 route show | jq -rM '.[] | select(.dst == "default")')" ]
+}
+
+# has_ipv6() - Check if one default route is available for IPv6
+function has_ipv6() {
+    [ -n "$(ip -j -6 route show | jq -rM '.[] | select(.dst == "default")')" ]
+}
+
+# skip_if_no_ipv4() - Skip current test if IPv4 traffic can't be routed
+# $1:	Optional message to display
+function skip_if_no_ipv4() {
+    if ! has_ipv4; then
+        local msg=$(_add_label_if_missing "$1" "IPv4")
+        skip "${msg:-not applicable with no routable IPv4}"
+    fi
+}
+
+# skip_if_no_ipv6() - Skip current test if IPv6 traffic can't be routed
+# $1:	Optional message to display
+function skip_if_no_ipv6() {
+    if ! has_ipv6; then
+        local msg=$(_add_label_if_missing "$1" "IPv6")
+        skip "${msg:-not applicable with no routable IPv6}"
+    fi
+}
+
+# has_pasta() - Check if the pasta(1) command is available
+function has_pasta() {
+    command -v pasta >/dev/null
+}
+
+# skip_if_no_pasta() - Skip current test if pasta(1) is not available
+# $1:	Optional message to display
+function skip_if_no_pasta() {
+    if ! has_pasta; then
+        local msg=$(_add_label_if_missing "$1" "pasta")
+        skip "${msg:-not applicable with no pasta binary}"
+    fi
+}
+
+
+### procfs access ##############################################################
+
+# ipv6_to_procfs() - RFC 5952 IPv6 address text representation to procfs format
+# $1:	Address in any notation described by RFC 5952
+function ipv6_to_procfs() {
+    local addr="${1}"
+    local pad
+    local nibble
+
+    case ${addr} in "::"*) addr=0"${addr}" ;; esac
+
+    case ${addr} in *"::"*)
+        pad="$(echo ":::::::::" | sed s/$(echo "${addr}" | sed s/[0-f]//g)//)"
+        pad="$(echo "${pad}" | sed s/:/:0/g)"
+        addr="$(echo "${addr}" | sed s/::/"${pad}"/g)"
+        ;;
+    esac
+
+    IFS=":"
```

Comment:
Yikes, no. One must never set `IFS` or other dangerous variables on a line by themselves. If anything fails or gets interrupted in the code below, you'd be left with a broken `IFS`, and very bad things would happen.

Reply:
Note that this is _not_ exported, rather _assigned_, so the scope is limited to a given test execution, as `bats` runs those in different shell processes. Strictly speaking, the scope is further limited (to exactly the next two lines) because an error in `printf` would cause the test to abort. I can indeed limit, explicitly, the scope to the `for` loop in question, though. I'll change this.

Reply:
Understood: the scope of danger is really small. But some future maintainers will see a single-line `IFS=` and feel very stressed out, needing to spend cycles (at high pulse rate) confirming that this is safe. As a courtesy to future maintainers, it's polite to just not do things like that.

Reply:
Gone anyway.